### PR TITLE
Fix for clang build and circleci

### PR DESCRIPTION
**Pull request by wep21**
https://github.com/ANYbotics/grid_map/pull/318 + Fix circleci.

### CODE REVIEW

**Comment by wep21** [comment]
@SteveMacenski This is ready for merge. Could you review this?

**Comment by wep21** [comment]
@SteveMacenski friendly ping.

**Comment by wep21** [comment]
@SteveMacenski friendly ping

File: tools/ros2_dependencies.repos [hunk 1]
```diff
@@ -1,13 +1,5 @@
 repositories:
```
Comment (SteveMacenski) [inline tools/ros2_dependencies.repos:1]: Are these removed because they're on released binaries now?

File: .circleci/config.yml [hunk 2]
```diff
@@ -23,15 +23,15 @@ jobs:
...
+            MAKEFLAGS="-j4 -l1" colcon build --symlink-install --executor sequential --packages-up-to nav2_costmap_2d
             source $UNDERLAY_WS/install/setup.bash
-            cd $OVERLAY_WS && rosdep install -y --ignore-src --from-paths src --skip-keys "octomap"
+            cd $OVERLAY_WS && DEBIAN_FRONTEND=noninteractive rosdep install -y --ignore-src --from-paths src
       - run:
           name: Debug Build
           command: |
             source $UNDERLAY_WS/install/setup.bash
-            cd $OVERLAY_WS && colcon build --parallel-workers 1
+            cd $OVERLAY_WS && MAKEFLAGS="-j4 -l1" colcon build --executor sequential
```
Comment (SteveMacenski) [inline .circleci/config.yml:33]: What's the value of this change?

File: tools/ros2_dependencies.repos (hunk 1 above)
Comment (wep21) [reply tools/ros2_dependencies.repos:1]: Yes, you're right.

File: .circleci/config.yml (hunk 2 above)
Comment (wep21) [reply .circleci/config.yml:33]: It's to reduce the ram usage for circle ci.

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**